src/Services/UserService.cs

```diff
@@ -32,6 +33,39 @@ public static async Task ListUsers(GraphServiceClient graphClient)
             }
         }
 
+        public static async Task ListUsersWithCustomAttribute(GraphServiceClient graphClient, string b2cExtensionAppClientId)
+        {
+            if (string.IsNullOrWhiteSpace(b2cExtensionAppClientId))
+            {
+                throw new ArgumentException("B2C Extension App ClientId (ApplicationId) is missing in the appsettings.json. Get it from the App Registrations blade in the Azure portal. The app registration has the name 'b2c-extensions-app. Do not modify. Used by AADB2C for storing user data.'.", nameof(b2cExtensionAppClientId));
+            }
+
+            // Declare the names of the custom attributes
+            const string favouriteSport = "FavouriteSport";
+            const string isAwesome = "IsAwesome";
```

Comment:
Consider naming these vars more generically, perhaps `customAttributeName1` and `customAttributeName2`, so as not to tie them so tightly to their (current) values.

Reply:
Renamed to more general but I made the complete name explicit otherwise very difficult to understand in your select statement against the graph...

Reply:
Good call.